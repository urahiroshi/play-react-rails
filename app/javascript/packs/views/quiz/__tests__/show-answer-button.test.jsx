import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ShowAnswerButton from '../show-answer-button'

describe('#render', () => {
  it('should show snapshot', () => {
    const wrapper = shallow(<ShowAnswerButton answer="hoge" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('when "Show Answer" button clicked', () => {
  it('should show answer', () => {
    const answer = 'hoge'
    const wrapper = shallow(<ShowAnswerButton answer={answer} />)
    expect(wrapper.contains(answer)).toBe(false)
    wrapper.find('button').props().onClick()
    expect(wrapper.contains(answer)).toBe(true)
  })
})
