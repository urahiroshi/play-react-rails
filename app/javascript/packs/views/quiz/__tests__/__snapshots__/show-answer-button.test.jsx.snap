// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should show snapshot 1`] = `
<div
  className="quiz-show-answer"
>
  <div>
    <button
      className="btn btn-warning quiz-answer-button"
      disabled={false}
      onClick={[Function]}
    >
      Show Answer
    </button>
  </div>
</div>
`;
